Detectors/TRD/base/src/TRDCommonParam.cxx

```diff
@@ -163,91 +165,71 @@ float TRDCommonParam::GetOmegaTau(float vdrift)
 //_____________________________________________________________________________
 bool TRDCommonParam::GetDiffCoeff(float& dl, float& dt, float vdrift)
 {
-  /*
   //
   // Calculates the diffusion coefficients in longitudinal <dl> and
   // transverse <dt> direction for a given drift velocity <vdrift>
   //
 
   // Nothing to do
-  if (TMath::Abs(fDiffLastVdrift - vdrift) < 1.e-3) {
-
-    dl = fDiffusionL;
-    dt = fDiffusionT;
+  if (TMath::Abs(mDiffLastVdrift - vdrift) < 0.001) {
+    dl = mDiffusionL;
+    dt = mDiffusionT;
     return true;
-
   }
 
-  fDiffLastVdrift = vdrift;
-
-  if      (IsXenon()) {
-
+  mDiffLastVdrift = vdrift;
+  if (IsXenon()) {
     //
     // Vd and B-field dependent diffusion and Lorentz angle
     //
 
     // The magnetic field strength
-    AliMagF* fld = (AliMagF *) TGeoGlobalMagField::Instance()->GetField();
+    o2::field::MagneticField* fld = static_cast<o2::field::MagneticField*>(TGeoGlobalMagField::Instance()->GetField());
```

Comment:
The way to extract the field is correct, but I would not repeat the very slow aliroot code: GetDiffCoeff is called a lot, so we better precalculate whatever is possible, see below.